### PR TITLE
[agent] remove potential data race on `sShouldTerminate`

### PR DESCRIPTION
**Pull request by simonlingoogle**
This commit remove potential data race on `sShouldTerminate`, which is accessed in both main loop and the signal handler.

### CODE REVIEW

File: src/agent/application.cpp [hunk 1]
```diff
@@ -43,8 +43,8 @@
 
 namespace otbr {
 
-bool                 Application::sShouldTerminate = false;
-const struct timeval Application::kPollTimeout     = {10, 0};
+std::atomic_bool     Application::sShouldTerminate{false};
```
Comment (jinran-google) [inline src/agent/application.cpp:46]: I think the old "= false" initialization is better.

**Review by jinran-google — APPROVED** [review]
LGTM overall 👍

File: src/agent/application.cpp (hunk 1 above)
Comment (simonlingoogle) [reply src/agent/application.cpp:46]: It does not compile.

Comment (jinran-google) [reply src/agent/application.cpp:46]: OK, it doesn't compile without mandatory copy elision, which was introduced in C++17.

Then I think it's better to use the traditional constructor syntax (with parentheses) in this case.

Comment (simonlingoogle) [reply src/agent/application.cpp:46]: > OK, it doesn't compile without mandatory copy elision, which was introduced in C++17.

Thanks for the insight. 


> Then I think it's better to use the traditional constructor syntax (with parentheses) in this case.

Is there a explanation why which is better?

**Review by bukepo — APPROVED** [review]

**Review by jwhui — APPROVED** [review]

**Pull request merged**